src/core/actions/cityactions/Build.java

```diff
@@ -1,46 +1,214 @@
 package core.actions.cityactions;
 
+import core.TechnologyTree;
+import core.TribesConfig;
 import core.Types;
 import core.actions.Action;
+import core.actors.Tribe;
+import core.actors.buildings.*;
+import core.game.Board;
 import core.game.GameState;
 import core.actors.City;
+import utils.Vector2d;
 
 import java.util.LinkedList;
 
 public class Build extends CityAction
 {
-    private Types.BUILDING building;
+    private Types.BUILDING buildingType;
 
     public Build(City c)
     {
         super.city = c;
     }
 
+    public void setBuildingType(Types.BUILDING buildingType) {this.buildingType = buildingType;}
+
+    public Types.BUILDING getBuildingType() {
+        return buildingType;
+    }
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute variants for a Build action
-        return new LinkedList<>();
+        LinkedList<Action> actions = new LinkedList<>();
+        Board board = gs.getBoard();
+        LinkedList<Vector2d> tiles = board.getCityTiles(city.getActorID());
+
+        for(Vector2d tile : tiles){
+            for(Types.BUILDING building: Types.BUILDING.values()){
+                //check if tile is empty
+                if(board.getBuildingAt(tile.x, tile.y) == null) {
+                    Build action = new Build(city);
+                    action.setBuildingType(building);
+                    action.targetPos = tile;
+                    if (action.isFeasible(gs)) {
+                        actions.add(action);
+                    }
+                }
+            }
+        }
+        return actions;
     }
 
     @Override
     public boolean isFeasible(final GameState gs) {
-        //TODO: Is feasible to build
+        //TODO: Add monuments
+        Tribe tribe = gs.getTribe(city.getTribeId());
+        Board board = gs.getBoard();
+        TechnologyTree t = tribe.getTechTree();
+        int stars = tribe.getStars();
+
+        switch (buildingType) {
```

Comment:
I think this switch can be refactored, extracting a function that checks if the building is "buildable". Something like

boolean isBuildable(int cost, Types.Building buildingType) {...}  //plus tribe and board, or game state, passed by parameter